rfqmath/fixed_point.go

```diff
@@ -103,6 +103,53 @@ func (f FixedPoint[T]) Equals(other FixedPoint[T]) bool {
 	return f.Coefficient.Equals(other.Coefficient) && f.Scale == other.Scale
 }
 
+// WithinTolerance returns true if the two FixedPoint values are within the
+// given tolerance (in parts per million (PPM)).
+func (f FixedPoint[T]) WithinTolerance(
```

Comment:
What's the intended use case of this method? To gauge if an action would result in precision loss?

Reply:
Looking at the tests, in what case would we ever be manipulating two values with different scale? If you look at the conversion methods, it handles scaling up and then down.

Reply:
> What's the intended use case of this method? To gauge if an action would result in precision loss?

This TODO describes one use case:
https://github.com/lightninglabs/taproot-assets/pull/1135/files#diff-64a27caf98eee653f23ffa833568a598241835b52d96fa4e26fc4127f92ad4f3R522

The current code uses this function:
```
pricesWithinBounds(firstPrice lnwire.MilliSatoshi,
	secondPrice lnwire.MilliSatoshi, tolerancePpm uint64) bool
```

I think we will need to replace that function with one that supports fixed-points.

Reply:
> Looking at the tests, in what case would we ever be manipulating two values with different scale? If you look at the conversion methods, it handles scaling up and then down.

`WithinTolerance` can handle fixed-point numbers of different scales. This feature may not always be necessary, but it's implemented to simplify usage, removing the need to scale arguments before calling the function.

We may need to handle fixed-point values at different scales in `pricesWithinBounds` callsites, depending on the values provided by our RFQ peer or the price oracle.